### PR TITLE
Skip limit_memory tests on Python 3.12

### LINKED ISSUE #14420: `ert-memory-tests` terminates due to segfault
```
Fatal Python error: Segmentation fault
...
...
...
Current thread 0x00007fb9c8c79740 (most recent call first):
  File "/tmp/f_scout_ci/actions-runner-02/_temp/source_root/tests/ert/performance_tests/test_dark_storage_performance.py", line 298 in test_plot_api_big_summary_memory_usage
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/python.py", line 167 in pytest_pyfunc_call
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/python.py", line 1707 in runtest
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/runner.py", line 184 in pytest_runtest_call
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/runner.py", line 250 in <lambda>
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/runner.py", line 361 in from_call
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/runner.py", line 249 in call_and_report
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/runner.py", line 139 in runtestprotocol
  File "/tmp/f_scout_ci/actions-runner-02/_temp/test-kenv/root/lib64/python3.12/site-packages/pytest_rerunfailures.py", line 968 in pytest_runtest_protocol
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/main.py", line 408 in pytest_runtestloop
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/main.py", line 384 in _main
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/main.py", line 330 in wrap_session
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/main.py", line 377 in pytest_cmdline_main
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/config/__init__.py", line 229 in _main
  File "/prog/res/komodo/bleeding-20260820-1256-py312-rhel8/root/lib64/python3.12/site-packages/_pytest/config/__init__.py", line 253 in _console_main
  File "/tmp/f_scout_ci/actions-runner-02/_temp/test-kenv/root/shims/pytest", line 10 in <module>

Extension modules: markupsafe._speedups, memray._memray, zmq.backend.cython._zmq, tornado.speedups, msgpack._cmsgpack, psutil._psutil_linux, charset_normalizer.md, charset_normalizer.cd, requests.packages.charset_normalizer.md, requests.packages.chardet.md, requests.packages.charset_normalizer.cd, requests.packages.chardet.cd, yaml._yaml, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, cftime._cftime, netCDF4._netCDF4, _cyutility, scipy._cyutility, scipy._lib._ccallback_c, shapely.lib, shapely._geos, shapely._geometry_helpers, xtgeo._cxtgeo, pyarrow.lib, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._generator, numpy.random._mt19937, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pyarrow._compute, pandas._libs.ops, numexpr.interpreter, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing, scipy.special._ufuncs_cxx, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg._batched_linalg, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._internal_matfuncs, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg.cython_lapack, scipy.linalg._decomp_update, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.interpolate._fitpack, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpacklib, scipy.sparse.linalg._propack, scipy.optimize._group_columns, scipy._lib.messagestream, scipy.optimize._trlib._trlib, sksparse.cholmod, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._slsqplib, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.spatial._ckdtree, scipy.spatial._hausdorff, scipy.spatial._distance_wrap, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial.transform._rotation_cy, scipy.spatial.transform._rigid_transform_cy, scipy.optimize._direct, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, scipy.ndimage._nd_image, scipy.ndimage._rank_filter_1d, _ni_label, scipy.ndimage._ni_label, h5py._errors, h5py.defs, h5py._objects, h5py.h5, h5py.utils, h5py.h5t, h5py.h5s, h5py.h5ac, h5py.h5p, h5py.h5r, h5py._npystrings, h5py._proxy, h5py._conv, h5py.h5z, h5py.h5a, h5py.h5d, h5py.h5ds, h5py.h5g, h5py.h5i, h5py.h5o, h5py.h5f, h5py.h5fd, h5py.h5pl, h5py.h5l, h5py._selector, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, gstools_cython.field, gstools_cython.krige, gstools_cython.variogram, scipy.special.cython_special, scipy.stats._stats, scipy.stats._biasedurn, scipy.stats._levy_stable.levyst, scipy.stats._stats_pythran, scipy.stats._ansari_swilk_statistics, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._rcont.rcont, scipy.stats._qmvnt_cy, lxml._elementpath, lxml.etree, PyQt6.QtCore, PyQt6.QtGui, PyQt6.QtWidgets, websockets.speedups, sklearn.__check_build._check_build, sklearn._cyutility, sklearn.utils._isfinite, sklearn.utils.sparsefuncs_fast, sklearn.utils.murmurhash, sklearn.utils._openmp_helpers, sklearn.utils._random, sklearn.utils._seq_dataset, sklearn.utils._cython_blas, sklearn.linear_model._cd_fast, sklearn.metrics.cluster._expected_mutual_info_fast, sklearn.metrics._dist_metrics, sklearn.metrics._pairwise_distances_reduction._datasets_pair, sklearn.metrics._pairwise_distances_reduction._base, sklearn.metrics._pairwise_distances_reduction._middle_term_computer, sklearn.utils._heap, sklearn.utils._sorting, sklearn.metrics._pairwise_distances_reduction._argkmin, sklearn.metrics._pairwise_distances_reduction._argkmin_classmode, sklearn.utils._vector_sentinel, sklearn.metrics._pairwise_distances_reduction._radius_neighbors, sklearn.metrics._pairwise_distances_reduction._radius_neighbors_classmode, sklearn.metrics._pairwise_fast, sklearn.preprocessing._csr_polynomial_expansion, sklearn.preprocessing._target_encoder_fast, sklearn._loss._loss, sklearn.utils.arrayfuncs, sklearn.linear_model._sag_fast, sklearn.svm._liblinear, sklearn.svm._libsvm, sklearn.svm._libsvm_sparse, sklearn.utils._weight_vector, sklearn.linear_model._sgd_fast, scipy.cluster.vq._vq, scipy.cluster.hierarchy._hierarchy, scipy.cluster.hierarchy._optimal_leaf_ordering, _cffi_backend, PyQt6.QtTest, PIL._imaging, kiwisolver._cext, PyQt6.QtSvg, PyQt6.QtSvgWidgets, grpc._cython.cygrpc, google._upb._message, pyarrow._parquet, pyarrow._fs, pyarrow._azurefs, pyarrow._hdfs, pyarrow._gcsfs, pyarrow._s3fs, multidict._multidict, yarl._quoting_c, propcache._helpers_c, aiohttp._http_writer, aiohttp._http_parser, aiohttp._websocket.mask, aiohttp._websocket.reader_c, frozenlist._frozenlist, pyarrow._acero, pyarrow._csv, pyarrow._json, pyarrow._substrait, pyarrow._dataset, pyarrow._dataset_orc, pyarrow._parquet_encryption, pyarrow._dataset_parquet_encryption, pyarrow._dataset_parquet (total: 254)
error: recipe `ert-memory-tests` was terminated on line 70 by signal 11
```

Possible first occurrence 2026-08-20;
https://github.com/equinor/komodo-releases/actions/runs/32361376662/job/96403373624
https://github.com/equinor/komodo-releases/actions/runs/32824379927/job/97731112326
https://github.com/equinor/komodo-releases/actions/runs/32853583737/job/97825653828
https://github.com/equinor/komodo-releases/actions/runs/33032681388/job/98389931251
https://github.com/equinor/komodo-releases/actions/runs/33458365017/job/99705384235
https://github.com/equinor/komodo-releases/actions/runs/33730025945/job/100570772473
https://github.com/equinor/komodo-releases/actions/runs/33746359464/job/100623043095
https://github.com/equinor/komodo-releases/actions/runs/34072434289/job/101593217968
https://github.com/equinor/komodo-releases/actions/runs/34451804193/job/102791073162
https://github.com/equinor/komodo-releases/actions/runs/34549494020/job/103111892359

### PR DESCRIPTION
**Pull request by erlenlh**
## Issue
Closes #14420

## Approach

`ert-memory-tests` has been failing intermittently with `SIGSEGV` (exit 139) since around 2026-08-20, both in this repo's CI and in komodo builds.

The crash is in memray, not in ert. Before [bloomberg/memray#1000](https://github.com/bloomberg/memray/pull/1000), `memray::compat::setprofileAllThreads` only used CPython's `PyEval_SetProfileAllThreads` for `PY_VERSION_HEX >= 0x030D0000` (3.13). On Python 3.12 it fell back to a hand-rolled loop over thread states, which races and crashes while the `Tracker` is torn down:

```
Py_XDECREF
_PyEval_SetProfile
memray::compat::setprofileAllThreads                            compat.cpp:17
memray::tracking_api::PythonStackTracker::removeProfileHooks    tracking_api.cpp:924
memray::tracking_api::Tracker::~Tracker                         tracking_api.cpp:924
```

Because a `SIGSEGV` kills the whole pytest process, the `@pytest.mark.flaky(reruns=5)` on `test_field_param_memory` cannot recover from it, and the entire `just ert-memory-tests` step aborts.

memray PR #1000 fixes this by changing the guard to `0x030C00F0`, but it is unreleased as of memray 1.20.0 (released 2026-08-07, merged 2026-08-28), so bumping `uv.lock` does not help. This PR therefore gates the `--memray` invocation on Python 3.13+.

Note that #14223 previously worked around the same crash by adding a `skipif` to `test_plot_api_big_summary_memory_usage`. That only moved the crash onto the other `limit_memory` test, which is why gating the invocation is preferred over another per-test skip.

`limit_memory` coverage still runs on Python 3.13 and 3.14 in CI.

Follow-up #14452 tracks removing the gate once a fixed memray is released.

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure unit tests pass locally after every commit (`git rebase -i main --exec 'just rapid-tests'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] New behavior or changes to existing untested code is covered by tests
- [ ] Commit messages follow the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by xjules — APPROVED** [review]
Don't forget to add ert issue to revert it back once memray will have got fixed.
🚀
